xql/main.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import xarray as xr
+import typing as t
+
+from sqlglot import parse_one, exp
+
+operate = {
+    "and" : lambda a, b: a & b,
+    "or" : lambda a, b: a | b,
+    "eq" : lambda a, b: a == b,
+    "gt" : lambda a, b: a > b,
+    "lt" : lambda a, b: a < b,
+    "gte" : lambda a, b: a >= b,
+    "lte" : lambda a, b: a <= b,
+}
+
+aggregate_function_map = {
+    'avg': lambda x, y: x.mean(dim=y) if y else x.mean(),
+    'min': lambda x, y: x.min(dim=y) if y else x.mean(),
+    'max': lambda x, y: x.max(dim=y) if y else x.mean(),
+}
+
+def parse(a, b):
+    arr_type = a.dtype.name
+    if arr_type == 'float64':
+        b = np.float64(b)
+    elif arr_type == 'float32':
+        b = np.float32(b)
+    elif arr_type == 'datetime64[ns]':
+        b = np.datetime64(b)
+    return a, b
```

Comment:
this parsing always assumes the LHS is the variable and RHS is the value (eg `temp > 5`)
general SQL query supports both `temp > 5` and `5 < temp`.
Additional checks before parsing  are required.

Or if XQL is not supporting `5 < temp` kind of conditions, should be mentioned in Docs.

Reply:
Added Support.